packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -174,6 +174,125 @@ class Plugin {
     );
   }
 
+  /// Queries the `pubpsec` and returns the `platforms` sub map.
+  ///
+  /// Throws ToolExit if cannot find the `platforms:` map.
+  static YamlMap getPlatformsYamlMap(YamlMap pubspec) {
+    if (pubspec == null) {
+       throwToolExit('Invalid pubspec.yaml', exitCode: 2);
+    }
+    final YamlMap flutterConfig = pubspec['flutter'] as YamlMap;
+    if (flutterConfig == null) {
+      throwToolExit('Did not find valid `flutter` map in pubspec.yaml', exitCode: 2);
+    }
+    final YamlMap pluginConfig = flutterConfig['plugin'] as YamlMap;
+    if (pluginConfig == null) {
+      throwToolExit('Did not find valid `plugin` map in pubspec.yaml', exitCode: 2);
+    }
+    if (pluginConfig['platforms'] == null) {
+      throwToolExit('Did not find valid `platforms` map in pubspec.yaml', exitCode: 2);
+    }
+
+    return pluginConfig['platforms'] as YamlMap;
+  }
+
+  /// Updates the pubspec.yaml under `projectDir`, adding each platform in `platforms` under `plugin: platforms:`
+  ///
+  /// If the dummy platform `some_platform` exists, remove it.
+  /// For example, if the original `plugin` map is:
+  /// ```
+  ///   flutter:
+  ///      plugin:
+  ///        platforms:
+  ///          some_platform:
+  ///            pluginClass: somePluginClass
+  /// ```
+  /// and the `platforms` is specified as `['ios','android'], the yaml map becomes:
+  /// ```
+  ///    flutter:
+  ///      plugin:
+  ///        platforms:
+  ///          android:
+  ///            package: <androidIdentifier>
+  ///            pluginClass: <pluginClass>
+  ///          ios:
+  ///            pluginClass: <pluginClass>
+  /// ```
+  ///
+  /// This method doesn't support the legacy plugin pubspec format and will throw error if try to update a pubspec.yaml that uses the legacy format.
+  static Future<void> updatePubspecWithPlatforms(String projectDir, List<String> platforms, String pluginClass, String androidIdentifier) async {
```

Comment:
It seems like there are a lot of ways for this method to fail, and it requires a sentinel data in the form of the comment. Is there a way this process could be simplified by ignoring comments? Is add-platforms going to be usable if it requires a very specific structure?

Reply:
one way to improve this is to ignore the lines that starts with a `#`

Other than that, I don't find a perfect solution to update this without breaking the existing yaml comments. Which I think we need to preserve because those are great instructions for new users.

Reply:
I would say ignore the comments and focus on the structured data.

Reply:
Did you mean to overwrite all the comments when updating pubspec.yaml?
Something like
`file.write(pubspecYamlMap.convertToString)`?
(I'm making up APIs here)

Or did you mean to skip comments like 
```
if (line.contains('#')) {
  continue;
}
```